### DISCUSSION TITLE
It's possible to reference resources outside of the root?

### DISCUSSION THREAD

**Discussion by ZeeD** [rokucommunity/brighterscript#769] [Q&A] [answered]
Hi.
I'm trying to split my project because I need to offer multiple "flavors" of my library.
while, on the "npm side" I can use workspaces, I would like to understand if I can tell `bsc` to support files outside of the project.
I have a layout like
```txt
 tree
.
├── flav1
│   ├── bsconfig.json
│   ├── config.json
│   └── package.json
├── flav2
│   ├── bsconfig.json
│   ├── config.json
│   ├── package.json
├── package.json
└── src
    ├── components
    ├── manifest
    └── source
        └── mysource.bs
```

focusing on `flav1` I would like to set, in the `bsconfig.json` something like
```json
{
    "rootDir": "../src",
    "files": [
        "manifest",
        "source/**/*.*",
        "components/**/*.*",
        "../flav1/config.json"
    ]
}
```
what I'm trying to tell is "use `../src` as the main folder of the project, but *also* include in the package the specific `config.json` I have here"

but `bsc` ends with the error `Error: Cannot reference a file outside of rootDir when using a top-level string. Please use a src;des; object instead`.
I've tried to put something like `"../flav1/config.json:config.json;"` in the `bsconfig.json` but it's not working.
Is there a way to tell `bsc` to include a file outside the `rootDir`?

**Comment by ZeeD** [comment, marked as answer]
... and I just found out that you can set ... 
```json
{
    "src": [
        "../flav1/config.json"
    ],
    "dest": "config.json"
}
```

**Reply by TwitchBronBron** [reply]
Yeah, this is the correct pattern. 

You can also reference entire folders with this pattern:
```jsonc
"src": "./", //this is your "flavor1/" folder
"files": [{
    "src": "../src/**/*",
    "dest": "" //start all relative paths at the root of the project, so "flavor1/"
}]
```
